src/screens/SSICredentialsRequiredScreen/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { InputDescriptorV1 } from '@sphereon/pex-models'
+import React, { FC, useEffect, useState } from 'react'
+import { ListRenderItemInfo } from 'react-native'
+import { SwipeListView } from 'react-native-swipe-list-view'
+import { PEX, SelectResults } from '@sphereon/pex'
+
+import { OVERVIEW_INITIAL_NUMBER_TO_RENDER } from '../../@config/constants'
+import SSICredentialRequiredViewItem from '../../components/views/SSICredentialRequiredViewItem'
+import {
+  SSIBasicContainerStyled as Container,
+  SSICredentialsRequiredScreenButtonContainerStyled as ButtonContainer,
+  SSIStatusBarDarkModeStyled as StatusBar
+} from '../../styles/components'
+import { ScreenRoutesEnum, StackParamList } from '../../types'
+import {
+  UniqueVerifiableCredential,
+  VerifiableCredential,
+} from '@veramo/core'
+import { getVerifiableCredentialsFromStorage } from '../../services/credentialService'
+import {
+  IVerifiableCredential,
+  OriginalVerifiableCredential
+} from '@sphereon/ssi-types'
+import SSIButtonsContainer from '../../components/containers/SSIButtonsContainer'
+import { translate } from '../../localization/Localization'
+
+type Props = NativeStackScreenProps<StackParamList, ScreenRoutesEnum.CREDENTIALS_REQUIRED>;
+
+const SSICredentialsRequiredScreen: FC<Props> = (props: Props): JSX.Element => {
+  const { presentationDefinition } = props.route.params
+  const [selectedCredentials, setSelectedCredentials] = useState(new Map<string, Array<VerifiableCredential>>())
+  const [availableCredentials, setAvailableCredentials] = useState(new Map<string, Array<VerifiableCredential>>())
+  const pex = new PEX()
+
+  useEffect(() => {
+    getVerifiableCredentialsFromStorage()
+      .then((vcs: Array<UniqueVerifiableCredential>) => {
+        const originalVcs: Array<OriginalVerifiableCredential> = vcs.map((vc: UniqueVerifiableCredential) => vc.verifiableCredential as OriginalVerifiableCredential)
+        const credentials = new Map<string, Array<VerifiableCredential>>()
+        presentationDefinition.input_descriptors.forEach((inputDescriptor: InputDescriptorV1) => {
+          const presentationDefinition = {
+            id: inputDescriptor.id,
+            input_descriptors: [inputDescriptor]
+          }
+          const selectResult: SelectResults = pex.selectFrom(presentationDefinition, originalVcs)
+          const matchedCredentials: Array<VerifiableCredential> = selectResult.verifiableCredential
+            ? selectResult.verifiableCredential.map((vc: IVerifiableCredential) => vc as VerifiableCredential)
+            : []
+          credentials.set(inputDescriptor.id, matchedCredentials)
+        })
+        setAvailableCredentials(credentials);
+      })
+  }, []);
+
+  useEffect(() => {
+    const credentials = new Map<string, Array<VerifiableCredential>>()
+    presentationDefinition.input_descriptors.forEach((inputDescriptor: InputDescriptorV1) => credentials.set(inputDescriptor.id, []))
```

Comment:
Hmzz, given the only direct difference between v1 and v2 input descriptors is the potential presence of the schema property (v1 only), IMO we should immediately accomodate for both at least in this screen. A simple union should suffice for now
Another difference is down the line in the constraints, but that is not something this screen is concerned about

Reply:
added v2 as well